src/widget/wtrackmenu.cpp

```diff
@@ -949,6 +967,12 @@ void WTrackMenu::slotOpenInFileBrowser() {
     mixxx::DesktopHelper::openInFileBrowser(locations);
 }
 
+void WTrackMenu::slotFindOnSoundcloud() {
+    const TrackPointer pTrack = getFirstTrackPointer();
+    QString artistName = pTrack->getArtist();
+    QDesktopServices::openUrl(QUrl("https://soundcloud.com/search/people?q=" + artistName));
```

Comment:
I recommend not to use naive string composition for creating URLs, see `QUrl::setQuery()`.

Please also externalize all the business logic into separate classes/files, don't bother `WTrackMenu` with this extra stuff. It is already too big.

Reply:
Thanks for the feedback, I will take a look at QUrl::setQuery() and work on business logic.

Reply:
Hey again,

I was trying to move all the logic into other classes and I was working on `QUrl::setQuery()`. I just have a question, 

I have a function, which takes 2 arguments one is the URL itself and the second one is query (in this case, Artist, Album artist, Album, Track title). My Url is `https://www.last.fm/search/artists?q=` and the artist is `Eminem`. 

When I call the method `QUrl::setQuery()`, it works just fine, the Url is created. When the Url is opened I realized the opened Url is `https://www.last.fm/search/artists?Eminem` which opens a page without any results. 'q=' part is gone. 

Should I pass all the queries with the prefix "=q" into my function or should I use another method which fits into key value pattern?

Because when I looked into documentation it says about 'setQuery'.
> This function is useful if you need to pass a query string that does not fit into the key-value pattern

Reply:
From the Qt doc, For me it seems like you should use [QUrlQuery](https://doc.qt.io/qt-5/qurlquery.html) and then use the `QUrl::setQuery(QUrlQuery)` overload. QUrlQuery:
> The QUrlQuery class provides a way to manipulate a key-value pairs in a URL's query. [More...](https://doc.qt.io/qt-5/qurlquery.html#details)

The QUrl class does much more than just concatenate string. When you pass it an a URL with a query and then call `setQuery`, it will overwrite the previous query, which is why `setQuery("Eminem")` results in the previous query (`q=` being overwritten). Since you are dealing with key-value pairs here, use `QUrlQuery`.

Reply:
Thanks for your feedback. 

Now I realized that I should use both the QUrlQuery and setQuery. For two different things.

If pagination is simple and I can directly open the artist-track-album page. I can simply use setQuery. Because there is no key and value pairs. There will be no queries. 

If we can't open artist-track-album page directly and if we need to do search on the service, I should use QUrlQuery, to have key q= and the value artist-track-album name.